### PR TITLE
changed graph annotation

### PR DESCRIPTION
**Pull request by Shania99**
Changed annotation text of output graph
pipeline.py takes new argument -d as path to json dump (assumes keys as pair-id)

### CODE REVIEW

**Comment by siv2r** [comment]
I have some doubts
- What does it do after reading the json dump?
- How will this data be incorporated after finding the match (i.e, reading from compatibility graph)?

**Review by siv2r — COMMENTED** [review]
Can you let me know which code is responsible for reading json dump?

**Comment by Shania99** [comment]
> * What does it do after reading the json dump?
Displays the name, age etc. 
> * How will this data be incorporated after finding the match (i.e, reading from compatibility graph)?
These details are displayed as node tooltips for the output graph

File: global_match/hovering.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+            20, 20), textcoords="offset points", bbox=dict(
+                boxstyle="round", fc="w"), arrowprops=dict(
+                    arrowstyle="->"))
+    annot.set_visible(False)
+    idx_to_node_dict = {}
+    for idx, node in enumerate(G.nodes):
+        idx_to_node_dict[idx] = node
+        
+    ### setting annotation text
+    def update_annot(ind, nodelist):
```
Comment (siv2r) [inline global_match/hovering.py:177]: can you add some documentation using the python docstring? It is quite difficult to understand what the variable and function do.

**Review by siv2r — CHANGES_REQUESTED** [review]

File: global_match/hovering.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+    pd_details : string -> path to JSON file (dump) with patient donor details
+     '''
+    # attrs = {}
+    # for node in G.nodes:
+    #     attrs[node] = {'Blood Group': 'A+', 'Pincode': 600001, 'Age': 33}
+    # nx.set_node_attributes(G, attrs)
+
+    fig, ax = plt.subplots()
+    pos = graphviz_layout(G)
+    data = read_details(pd_details)
```
[the remark this answers is not in the document]
Comment (Shania99) [reply global_match/hovering.py:31]: `data` is the dictionary that contains the patient details. It is used to display the details in the annotation box

File: global_match/hovering.py (hunk 1 above)
Comment (Shania99) [reply global_match/hovering.py:177]: Sure I'll do that

File: global_match/hovering.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (siv2r) [reply global_match/hovering.py:31]: where is the annotation box in this function? Can you add a comment on that line?

File: global_match/hovering.py [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+    nx.draw_networkx_edge_labels(G, pos, edge_labels=w_rest, label_pos=0.5)
+
+    # =================== HOVERING =========================
+    ### setting annotation style
+    annot = ax.annotate(
+        "", xy=(
+            0, 0), xytext=(
+            20, 20), textcoords="offset points", bbox=dict(
+                boxstyle="round", fc="w"), arrowprops=dict(
+                    arrowstyle="->"))
```
Comment (Shania99) [inline global_match/hovering.py:170]: This is the annotation box

File: global_match/hovering.py [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+        node = list(nodelist)[node_idx]
+        xy = pos[node]
+        annot.xy = xy
+        node_attr = {'ID': node}
+        node_attr.update(G.nodes[node])
+        all_details = data[node]
+        patient_string = 'Patient: {} , {}, {}'.format(all_details["pName"], all_details["pBgrp"], all_details["pAge"])
+        donor_string = 'Donor: {} , {}, {}'.format(all_details["dName"], all_details["dBgrp"], all_details["dAge"])
+        text = '\n'.join([patient_string,donor_string])
+        annot.set_text(text)
```
Comment (Shania99) [inline global_match/hovering.py:188]: This is the text of the annotation box

**Pull request merged**